# Introduction
In this project I'm tasked with identifying and proving which is the "best" neighborhood in Pittsburgh using a data set that's collected from each neighborhood in Pittsburgh. I was originally going to use a metric that collected SNAP data from each neighborhood or a metric that reported landslide prone areas, but I ultimately chose a metric based on the monthly criminal activity in each neighborhood as my group and I decided to chose metrics based on safety.


# Dataset: Monthly Criminal Activity

In [ ]:
import pandas as pd

In [ ]:
crime = pd.read_csv("/ihome/cmpinf0010_2025f/ajm698/Final-Project-Best-Pittsburgh-Neighborhood-/data/Monthly_Criminal_Activity.csv", low_memory=False)


In [ ]:
#cf = crime filter
cf = crime[["Neighborhood", "ReportedMonth", "NIBRS_Offense_Type"]]


In [ ]:
#each neighborhood
neighborhoods = crime["Neighborhood"].unique()
neighborhoods_list = []

for neighborhood in neighborhoods:
    if isinstance(neighborhood, str):
        neighborhoods_list.append(neighborhood)

for neighborhood in neighborhoods_list:
    print(neighborhood)


In [ ]:
#each offense type
offense_type = crime["NIBRS_Offense_Type"].unique()
offense_type_list = []

for offense in offense_type:
    if isinstance(offense, str):
        offense_type_list.append(offense)

offense_type_list.sort()

for offense in offense_type_list:
    print(offense)


In [ ]:
#total crimes each neighborhood has for each month
#avg crimes per month for each neighboorhood
#total crimes per month per neighborhood 
cf = cf.copy()

cf_monthly = cf.groupby(["Neighborhood", "ReportedMonth"])["NIBRS_Offense_Type"].count()
avg_monthly = cf_monthly.groupby("Neighborhood").mean()

In [ ]:
#the total crimes each neighborhood has had
#avg crimes for this year for each neighboorhood
cf_year = cf_monthly.groupby("Neighborhood").sum()

In [ ]:
#neighborhoods_list = []
ndata = []

for neighborhood in neighborhoods_list:
    ndata.append({"Neighborhood": neighborhood, "Average_Crime_Per_Month": avg_monthly.loc[neighborhood], "Total_Crime_This_Year": cf_year.loc[neighborhood]})

data = pd.DataFrame(ndata)
data

In [62]:
lowest5_monthly = data.nsmallest(5, "Average_Crime_Per_Month")[["Neighborhood", "Average_Crime_Per_Month"]]

lowest5monthly = pd.DataFrame({
    "Neighborhoods": lowest5_monthly["Neighborhood"].values,
    "Lowest5_Monthly_Avg": lowest5_monthly["Average_Crime_Per_Month"].values
                          })
lowest5monthly.head()

Neighborhoods  Lowest5_Monthly_Avg
0       Chartiers             3.166667
1       Ridgemont             3.300000
2     Saint Clair             3.571429
3   New Homestead             5.400000
4  Swisshelm Park             6.000000

In [63]:
lowest5_yearly = data.nsmallest(5, "Total_Crime_This_Year")[["Neighborhood", "Total_Crime_This_Year"]]

lowest5year = pd.DataFrame({
    "Neighborhoods": lowest5_yearly["Neighborhood"].values,
    "Lowest5_This_Year": lowest5_yearly["Total_Crime_This_Year"].values
                          })
lowest5year.head()

Neighborhoods  Lowest5_This_Year
0     Saint Clair                 25
1       Ridgemont                 33
2       Chartiers                 38
3   New Homestead                 54
4  Swisshelm Park                 66

# Conclusion
According to my data I believe Chartiers is the best neighborhood as Chartiers had the lowest crime average per month at ≈ 3.17 crimes. Based on this data Chartiers is shown to be the safest neigbhorhood due to it's low monthly crime rate and in the context of my metric it's proved to be the best neighborhood in Pittsburgh. The data driven determination that Chartiers is the "best neighborhood" is completely different from my personal favorite neighborhoods, Oakland and around Allgheny West/Perry South. This is due to the fact that I've never heard or even been to Chartiers, compared to Oakland where Pitt is located and Allgheny West/Perry South where I grew up going to, to visit family.